client/src/wayland_obj/wl_shm.rs

```diff
@@ -1,59 +1,90 @@
 //! Wrapper around a wl_shm.
 
-use std::{cell::RefCell, os::unix::io::RawFd};
+use std::{fs::File, io::Write};
 
 use wayland_client::{
-    self,
     protocol::{
         wl_buffer::WlBuffer,
-        wl_shm::{self, WlShm}
+        wl_shm::{Format, WlShm}
     },
     NewProxy
 };
 
 use crate::area::Size;
 
-/// The minimum version of the wl_shm global to bind to.
-pub const WL_SHM_VERSION: u32 = 1;
+// TODO(ried): the WlBuffer internal should be internal and Buffer should be Clone
```

Comment:
This would require cloning the File handle. std::fs::File provides a try_clone method, any concerns with this approach?

Reply:
Why does this need to be clone? If so, then does it make sense that both buffers refer to the same file/contents, even though they are "different" buffers?

I think instead that a new temp file should be created on `clone`. However, if we don't need this to be clone it should not be because it's going to be expensive to do so.

If whatever needs to clone this is instead trying to get a reference to this buffer either borrows or `Rc` should be used.